includes/html/api_functions.inc.php

```diff
@@ -1844,6 +1844,34 @@ function rename_device(Illuminate\Http\Request $request)
     }
 }
 
+function update_device_attrib(Illuminate\Http\Request $request)
+{
+    $hostname = $request->route('hostname');
+    // use hostname as device_id if it's all digits
+    $device_id = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+    $data = json_decode($request->getContent(), true);
+    if (empty($data['attrib_type'])) {
+        return api_error(400, 'Device Attrib field to patch has not been supplied');
+    }
+    if (is_array($data['attrib_type']) && is_array($data['attrib_value'])) {
+        if (count($data['attrib_type']) == count($data['attrib_value'])) {
+            for ($x = 0; $x < count($data['attrib_type']); $x++) {
+                if (dbUpdate(['attrib_value' => $data['attrib_value'][$x]], 'devices_attribs', '`device_id`=? and `attrib_type`=?', [$device_id, $data['attrib_type'][$x]]) < 0) {
```

Comment:
Please don't use the legacy db commands.  https://laravel.com/docs/8.x/eloquent
The Device model also has attrib helpers on it.

Reply:
To be honest, I started with a copy of the update_device API, and adapted it from there.  The device helpers for attribs is a better spot to work from.  I'll look at re-writing for that.